src/components/ArticleList.jsx

```diff
@@ -1,71 +1,111 @@
 import { useEffect } from "react"
 import { useContext } from "react"
 import { useState } from "react"
-import { getAllArticles, getAllTopics, getArticleByTopic } from "../api"
+import { useNavigate } from "react-router-dom";
+
+import { getAllArticles, getAllTopics, getArticleByQuery } from "../api"
 import { TopicContext } from "../contexts/TopicContext"
 import { TopicListContext } from "../contexts/TopicListContext"
 import { ArticleCard } from "./ArticleCard"
 
 export const Articles = ()=> {
     const[articlesList, setArticlesList] = useState([])
     const[isLoading , setIsLoading] = useState(true)
+    const[queries, setQueries] = useState('')
     const {topic, setTopic} = useContext(TopicContext)
     const {topicList, setTopicList} = useContext(TopicListContext)
-
+    const navigate = useNavigate()
     
-
     useEffect(()=> {
         setIsLoading(true)
         if(topic.length) {
-            getArticleByTopic(topic).then((articles)=> {
+            getArticleByQuery(`topic=${topic}`).then((articles)=> {
                 setArticlesList(articles)
                 setIsLoading(false)
             })
-        } else {
+        } else{
            getAllArticles().then((articles) => {
             setArticlesList(articles)
             setIsLoading(false)
         }) 
         }
+        if(queries.length){
```

Comment:
This if block and the previous if/else block are **both** running, meaning setArticlesList is being set twice, resulting in the articles not always being sorted.